uamqp/_connection.py

```diff
@@ -222,14 +253,29 @@ def _get_next_outgoing_channel(self):
             raise ValueError("Maximum number of channels ({}) has been reached.".format(self._channel_max))
         next_channel = next(i for i in range(1, self._channel_max) if i not in self._outgoing_endpoints)
         return next_channel
-    
+
     def _outgoing_empty(self):
         # type: () -> None
         """Send an empty frame to prevent the connection from reaching an idle timeout."""
         self._last_frame_sent_time = time.time()
         if self._network_trace:
             _LOGGER.info("-> empty()", extra=self._network_trace_params)
-        self._transport.write(EMPTY_FRAME)
+        try:
+            raise self._error
+        except TypeError:
+            pass
+        try:
+            if self._can_write():
+                self._transport.write(EMPTY_FRAME)
+                self._last_frame_sent_time = time.time()
```

Comment:
Maybe we should remove line 260, as it seems to be a duplicate

Reply:
good catch